test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -321,15 +322,30 @@ func AccumulateNames(nrts []nrtv1alpha2.NodeResourceTopology) sets.String {
 	return nodeNames
 }
 
-func FilterTopologyManagerPolicy(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy nrtv1alpha2.TopologyManagerPolicy) []nrtv1alpha2.NodeResourceTopology {
+func FilterByTopologyManagerPolicy(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy string) []nrtv1alpha2.NodeResourceTopology {
+	return filterByAttribute(nrts, nrtv1alpha2.AttributeInfo{Name: e2enrt.TopologyManagerPolicyAttribute, Value: tmPolicy})
+}
+
+func FilterByTopologyManagerScope(nrts []nrtv1alpha2.NodeResourceTopology, tmScope string) []nrtv1alpha2.NodeResourceTopology {
+	return filterByAttribute(nrts, nrtv1alpha2.AttributeInfo{Name: e2enrt.TopologyManagerScopeAttribute, Value: tmScope})
+}
+
+func FilterByTopologyManagerPolicyAndScope(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy, tmScope string) []nrtv1alpha2.NodeResourceTopology {
+	return FilterByTopologyManagerScope(FilterByTopologyManagerPolicy(nrts, tmPolicy), tmScope)
+}
+
+func filterByAttribute(nrts []nrtv1alpha2.NodeResourceTopology, attrInfo nrtv1alpha2.AttributeInfo) []nrtv1alpha2.NodeResourceTopology {
 	ret := []nrtv1alpha2.NodeResourceTopology{}
 	for _, nrt := range nrts {
-		if !contains(nrt.TopologyPolicies, string(tmPolicy)) {
-			klog.Warningf("SKIP: node %q doesn't support topology manager policy %q", nrt.Name, string(tmPolicy))
+		attr, ok := attribute.Get(nrt.Attributes, attrInfo.Name)
+		if !ok {
+			klog.Error("SKIP: node %q doesn't have required attribute %q", nrt.Name, attrInfo.Name)
+			continue
+		}
+		if attr.Value != attrInfo.Value {
+			klog.Warningf("SKIP: node %q doesn't wrong attribute value actual:%q, expected: %q", nrt.Name, attr.Value, attrInfo.Value)
 			continue
 		}
-		klog.Infof("ADD : node %q supports topology manager policy %q", nrt.Name, string(tmPolicy))
-		ret = append(ret, nrt)
```

Comment:
the `append` went missing and should be restored. To prevent this to occurr in the future we can move code in `internal/noderesourcetopologies` and add unit tests (possibly in a future PR)

Reply:
Thanks for the catch ... 
Yeah need to add some unit testing to avoid this. I will try to keep that in mind in the future.